deltachat-ffi/deltachat.h

```diff
@@ -2588,6 +2594,85 @@ char* dc_get_last_error (dc_context_t* context);
 void dc_str_unref (char* str);
 
 
+/**
+ * @class dc_backup_sender_t
+ *
+ * Set up another device.
+ */
+
+/**
+ * Create an object for sending a backup to another device.
+ * The backup is sent through an peer-to-peer channel which is bootstrapped by a QR-code.
+ * As a backup contains the whole state of the account, including credentials,
+ * this can be used to setup a new device.
+ *
+ * @memberof dc_backup_sender_t
+ * @param context The context.
+ * @param folder Path to create the backup in before it is sent out.
+ *     The path must exist and being accessible at least until the other device has received all data.
+ *     The folder is not cleaned up by the backup send;
+ *     a temporary directory seems to be good place therefore.
+ * @param passphrase Used to at-rest-encrypt the backuped database in `folder` before sending,
+ *     similar to dc_imex().
+ *     The same passphrase must be given to dc_receive_backup() on the other device.
+ *     If NULL or empty string is given, the sent backup is not encrypted at rest.
+ *     The transport of the backup is always encrypted additionally.
+ * @return Opaque object for sending the backup.
+ *     On errors, NULL is returned and dc_get_last_error() returns an error that should be shown to the user.
+ */
+dc_backup_sender_t* dc_send_backup (dc_context_t* context, const char* folder, const char* passphrase);
+
+
+/**
+ * Get QR code text that will offer the backup to other devices.
+ * The QR code contains a ticket that will unambiguously identify the backup.
+ *
+ * The scanning device will pass the scanned content to dc_check_qr() then;
+ * if dc_check_qr() returns DC_QR_BACKUP,
+ * the backup transfer can be started using dc_receive_backup().
+ *
+ * @memberof dc_backup_sender_t
+ * @param context The context.
+ * @param backup_sender The backup sender object as created by dc_send_backup().
+ * @return The text that should go to the QR code,
+ *     On errors, an empty QR code is returned, NULL is never returned.
+ *     The returned string must be released using dc_str_unref() after usage.
+ */
+char* dc_backup_sender_qr (dc_context_t* context, const dc_backup_sender_t* backup_sender);
+
+
+/**
+ * Waits for the sending to finish and frees the backup sender object.
+ *
+ * @memberof dc_backup_sender_t
+ * @param context The context.
+ * @param backup_sender The backup sender object as created by dc_send_backup(),
+ *     If NULL is given, nothing is done.
+ */
+void dc_backup_sender_done (dc_context_t* context, dc_backup_sender_t* backup_sender);
+
+
+/**
+ * Receive a backup offered by a dc_backup_sender_t object on another device.
+ *
+ * This function is typically not called on the device where the dc_backup_sender_t object exist,
+ * but on a device that has scanned the QR code generated by dc_backup_sender_qr().
+ *
+ * While dc_receive_backup() returns immediately, the started job may take a while;
+ * you can stop it using dc_stop_ongoing_process().
```

Comment:
An ongoing process is IIRC a global modal that can not be dismissed.  I guess this is normally done from the "create account" screen or similar so that is what is desired?